Library/Homebrew/cmd/audit.rb

```diff
@@ -1160,20 +1160,31 @@ def audit_urls
            %r{^http://mirrors\.kernel\.org/},
            %r{^http://(?:[^/]*\.)?bintray\.com/},
            %r{^http://tools\.ietf\.org/},
-           %r{^http://www\.mirrorservice\.org/},
            %r{^http://launchpad\.net/},
            %r{^http://bitbucket\.org/},
+           %r{^http://cpan\.metacpan\.org/},
            %r{^http://hackage\.haskell\.org/},
            %r{^http://(?:[^/]*\.)?archive\.org},
-           %r{^http://(?:[^/]*\.)?freedesktop\.org}
+           %r{^http://(?:[^/]*\.)?freedesktop\.org},
+           %r{^http://(?:[^/]*\.)?mirrorservice\.org/}
         problem "Please use https:// for #{p}"
-      when %r{^http://search\.(mcpan|cpan)\.org/CPAN/(.*)}i
-        problem "#{p} should be `https://cpan.metacpan.org/#{$2}`"
+      when %r{^http://search\.mcpan\.org/CPAN/(.*)}i
+        problem "#{p} should be `https://cpan.metacpan.org/#{$1}`"
       when %r{^(http|ftp)://ftp\.gnome\.org/pub/gnome/(.*)}i
         problem "#{p} should be `https://download.gnome.org/#{$2}`"
       end
     end
 
+    # Prefer HTTP/S when possible over FTP protocol due to possible firewalls.
+    urls.each do |p|
+      if p.start_with? "ftp://ftp.mirrorservice.org"
+        problem "Please use https:// for #{p}"
+      end
+      if p =~ %r{^ftp://ftp\.cpan\.org/pub/CPAN(.*)}i
```

Comment:
I'd probably make this `elsif` and fuse it with the preceding `if` block, as these cases are mutually exclusive. Speaking of `case`s, maybe worth making this a `case`-`when` construct to mirror the structure of the preceding `urls.each do |p|` block?